src/swifft.c

```diff
@@ -392,4 +392,15 @@ void SWIFFT_ComputeMultipleSigned(int nblocks, const BitSequence * input,
 	SWIFFT_ISET_NAME(SWIFFT_ComputeMultipleSigned_)(nblocks, input, sign, output);
 }
 
+//! \brief Computes the result of a SWIFFT operation and returns the value. Made to be compatible with Python.
+//! The result is composable with other hash values.
+//!
+//! \param[in] input the input of 256 bytes (2048 bit).
+BitSequence SWIFFT_Hash(const BitSequence input[SWIFFT_INPUT_BLOCK_SIZE])
+{
+	SWIFFT_ALIGN BitSequence result[SWIFFT_OUTPUT_BLOCK_SIZE];
```

Comment:
This is a bug as then the function returns a pointer to a stack-allocated variable, which is undefined behavior (see [this SO](https://stackoverflow.com/questions/39373525/c-function-returning-a-pointer-to-a-local-variable), for example). While this can be fixed by dynamically allocating the buffer here, don't do this because LibSWIFFT does not do dynamic allocations by design. Instead, create the needed buffers in the wrapper code and pass them to SWIFFT_Compute.

Reply:
Oh wow, this was a mistake commit after I opened the PR. Explains the problems I was having, I just didn't understand it! I've removed the code in the latest commit.